src/core/js/views/notifyView.js

```diff
@@ -1,268 +1,262 @@
-    }
+  attributes () {
+    return Object.assign({
+      role: 'dialog',
+      'aria-labelledby': 'notify-heading',
+      'aria-modal': 'true'
+    }, this.model.get('_attributes'));
+  }
 
-    setupEventListeners() {
-      this.listenTo(Adapt, {
-        'remove page:scrollTo': this.closeNotify,
-        'notify:resize': this.resetNotifySize,
-        'notify:cancel': this.cancelNotify,
-        'notify:close': this.closeNotify,
-        'device:resize': this.resetNotifySize
-      });
-      this.setupEscapeKey();
-    }
+  events() {
+    return {
+      'click .js-notify-btn-alert': 'onAlertButtonClicked',
+      'click .js-notify-btn-prompt': 'onPromptButtonClicked',
+      'click .js-notify-close-btn': 'onCloseButtonClicked',
+      'click .js-notify-shadow-click': 'onShadowClicked'
+    };
+  }
 
-    setupEscapeKey() {
-      $(window).on('keyup', this.onKeyUp);
-    }
+  initialize() {
+    _.bindAll(this, 'resetNotifySize', 'onKeyUp');
+    this.disableAnimation = Adapt.config.get('_disableAnimation') || false;
+    this.isOpen = false;
+    this.hasOpened = false;
+    this.setupEventListeners();
+    this.render();
+  }
 
-    onKeyUp(event) {
-      if (event.which !== 27) return;
-      event.preventDefault();
-      this.cancelNotify();
-    }
+  setupEventListeners() {
+    this.listenTo(Adapt, {
+      'remove page:scrollTo': this.closeNotify,
+      'notify:resize': this.resetNotifySize,
+      'notify:cancel': this.cancelNotify,
+      'notify:close': this.closeNotify,
+      'device:resize': this.resetNotifySize
+    });
+    this.setupEscapeKey();
+  }
 
-    render() {
-      const data = this.model.toJSON();
-      const template = Handlebars.templates.notify;
-      // hide notify container
-      this.$el.css('visibility', 'hidden');
-      // attach popup + shadow
-      this.$el.html(template(data)).appendTo('body');
-      // hide popup
-      this.$('.notify__popup').css('visibility', 'hidden');
-      // show notify container
-      this.$el.css('visibility', 'visible');
-      this.showNotify();
-      return this;
-    }
+  setupEscapeKey() {
+    $(window).on('keyup', this.onKeyUp);
+  }
 
-    onAlertButtonClicked(event) {
-      event.preventDefault();
-      // tab index preservation, notify must close before subsequent callback is triggered
-      this.closeNotify();
-      Adapt.trigger(this.model.get('_callbackEvent'), this);
-    }
+  onKeyUp(event) {
+    if (event.which !== 27) return;
+    event.preventDefault();
+    this.cancelNotify();
+  }
 
-    onPromptButtonClicked(event) {
-      event.preventDefault();
-      // tab index preservation, notify must close before subsequent callback is triggered
-      this.closeNotify();
-      Adapt.trigger($(event.currentTarget).attr('data-event'), this);
-    }
+  render() {
+    const data = this.model.toJSON();
+    const template = Handlebars.templates.notify;
+    // hide notify container
+    this.$el.css('visibility', 'hidden');
```

Comment:
could add and remove the `u-visibility-hidden` class based on if the element should be displayed.

Reply:
This is rendering effect to produce DOM area and no flashes whilst rendering, the `visibility` value is only set and unset in this function.